opentech/apply/funds/models.py

```diff
@@ -41,11 +51,30 @@ def get_defined_fields(self):
     def workflow_class(self):
         return WORKFLOW_CLASS[self.get_workflow_display()]
 
+    @property
+    def open_round(self):
+        rounds = Round.objects.child_of(self).live().public().specific()
+        return rounds.filter(
+            Q(start_date__lte=date.today()) &
+            Q(Q(end_date__isnull=True) | Q(end_date__gte=date.today()))
+        ).first()
+
+    def next_deadline(self):
+        return self.open_round.end_date
+
     content_panels = AbstractStreamForm.content_panels + [
         FieldPanel('workflow'),
         InlinePanel('forms', label="Forms"),
     ]
 
+    def serve(self, request):
+        if hasattr(request, 'is_preview'):
+            return super().serve(request)
+
+        # delegate to the open_round to use the latest form instances
+        request.show_page = True
+        return self.open_round.serve(request)
```

Comment:
The result of this is that after one submits the application form, they are taken to the round page which returns a 404

Reply:
happy this was sorted by e77d60b37